### ISSUE TITLE
[a11y] SidePanel: toggle button aria-labelledby reference is broken when collapsed

### ISSUE THREAD

**Issue by williamjstanton** [Workday/canvas-kit#2523]
## 🐛 Bug Report

When SidePanel component is in a collapsed state, the ToggleButton sub-component is still using the same aria-labelledby `id` reference to the heading that can no long be found in the DOM. 

This issue can also be observed on the `aria-labelledby` reference on the `<section>` element. 

## To Reproduce

All browsers, all screen readers. 
1. Turn on VoiceOver (macOS), launch Safari
2. Navigate to [Storybook Basic Example](https://workday.github.io/canvas-kit/?path=/docs/preview-side-panel--basic)
3. Set rotor to "landmarks": Ctrl + Opt + Cmd + RightArrow
4. Jump to landmark "Tasks Panel, region": Ctrl + Opt + Cmd + DownArrow
5. Observe: VoiceOver finds the Side Panel Basic example
6. Read Toggle Button: Ctrl + Opt + RightArrow
7. Observe: "Tasks Panel, expanded button"
8. Activate button: Ctrl + Opt + SpaceBar
9. Read backward to previous heading: Ctrl + Opt + Cmd + Shift + h
10. Repeat STEP 4: Jump to landmark 
11. (ISSUE) Observe VoiceOver cannot find landmark anymore
12. (ISSUE) Observe toggle button is now described as "collapsed button"

## Expected Behavior

The Side Panel landmark region should work regardless of expanded / collapsed state. The `<section>` element must have a valid accessible name. 

The Toggle Button must have a valid accessible name. 

## Actual Results

A clear and concise description of what actually happens

Please refer to steps 11, 12 in the reproduction steps. Refer to attached screen shots. 

## Browser (if applicable)

If this was a visual bug, what browser did you spot it on?

All of them. 

## Link to repl or repo (highly encouraged)

Please provide a reference to a repo or branch to help us reproduce this bug.
[Storybook Basic Example](https://workday.github.io/canvas-kit/?path=/docs/preview-side-panel--basic) 

Include a screenshot of a visual bug
![ck - SidePanel ToggleButton missing label](https://github.com/Workday/canvas-kit/assets/40372497/bdabc91c-2454-4de5-9c00-a76eb9f187b5)

![ck - SidePanel section missing name](https://github.com/Workday/canvas-kit/assets/40372497/d4799595-4bbd-4d9a-abed-e2c86919d34c)

**Comment by williamjstanton** [comment]
It occurs to me with the release of the new <AccessibleHide> component, I wonder if that can be used in the SidePanel component when it is in a collapsed state. This would preserve the 'aria-labelledby' id reference even while the element is not visible on screen.

**Comment by williamjstanton** [comment]
> It occurs to me with the release of the new AccessibleHide component, I wonder if that can be used in the SidePanel component when it is in a collapsed state. This would preserve the 'aria-labelledby' id reference even while the element is not visible on screen.

modules/react/side-panel/lib/SidePanel.tsx

Note to self, I think on line 216 you could return the AccessibleHide component instead of null, and maybe that'll fix this without changing any of the aria-labelledby references...?
